ethereum/spec/src/main/java/tech/pegasys/teku/spec/datastructures/blobs/versions/deneb/BlobSidecar.java

```diff
@@ -84,26 +87,26 @@ public SignedBeaconBlockHeader getSignedBeaconBlockHeader() {
     return getField4();
   }
 
-  public List<Bytes32> getKzgCommitmentInclusionProof() {
-    return getField5().stream().map(AbstractSszPrimitive::get).toList();
+  public SszBytes32Vector getKzgCommitmentInclusionProof() {
+    return getField5();
   }
 
   public UInt64 getSlot() {
     return getSignedBeaconBlockHeader().getMessage().getSlot();
   }
 
-  public Bytes32 getBlockRoot() {
-    return getSignedBeaconBlockHeader().getMessage().getRoot();
+  public Bytes32 getBlockBodyRoot() {
+    return getSignedBeaconBlockHeader().getMessage().getBodyRoot();
   }
 
   public SlotAndBlockRoot getSlotAndBlockRoot() {
-    return new SlotAndBlockRoot(getSlot(), getBlockRoot());
+    return new SlotAndBlockRoot(getSlot(), getBlockBodyRoot());
   }
 
   public String toLogString() {
     return LogFormatter.formatBlobSidecar(
         getSlot(),
-        getBlockRoot(),
+        getBlockBodyRoot(),
```

Comment:
Nice catch, which puts a focus on another issue: we cannot save it to DB since this change unless we get a block with corresponding `body_root`.
I have added both `body_root` and `state_root` to log string, since on most explorers we don't have `body_root`, but we may still want to see it to identify corresponding block.

Reply:
I'm stupid, we could do block root from block header. All fixed